src/braket/aws/aws_emulator_helpers.py

```diff
@@ -0,0 +1,201 @@
+        verbatim circuits only use native gates.
+    """
+
+    if isinstance(properties, IqmDeviceCapabilities):
+        try:
+            supported_gates.remove("start_verbatim_box")
+            supported_gates.remove("end_verbatim_box")
+        except ValueError:
+            pass
+
+    native_gates = properties.paradigm.nativeGateSet
+
+    return GateCriterion(supported_gates=supported_gates, native_gates=native_gates)
+
+
+@singledispatch
+def create_connectivity_criterion(
+    properties: DeviceCapabilities, connectivity_graph: DiGraph
+) -> ConnectivityCriterion:
+    """
+    Creates a ConnectivityCriterion pass which validates that multi-qubit gates are applied to
+    connected qubits based on this device's connectivity graph.
+
+    Args:
+        properties (DeviceCapabilities): QPU Device Capabilities object with a
+            QHP-specific schema.
+
+        connectivity_graph (DiGraph): Connectivity graph for this device.
+
+    Returns:
+        ConnectivityCriterion: An emulator pass that checks that a circuit only applies two-qubit
+        gates to connected qubits on the device.
+    """
+    connectivity_criterion = ConnectivityCriterion(connectivity_graph)
+    return connectivity_criterion
+
+
+@create_connectivity_criterion.register(IqmDeviceCapabilities)
+def _(properties: IqmDeviceCapabilities, connectivity_graph: DiGraph) -> ConnectivityCriterion:
+    """
+    IQM qubit connectivity is undirected but the directed graph that represents qubit connectivity
+    does not include back-edges. Thus, we must explicitly introduce back edges before creating
+    the ConnectivityCriterion for an IQM device.
+    """
+    connectivity_graph = connectivity_graph.copy()
+    for edge in connectivity_graph.edges:
+        connectivity_graph.add_edge(edge[1], edge[0])
+    return ConnectivityCriterion(connectivity_graph)
+
+
+@singledispatch
+def create_gate_connectivity_criterion(
+    properties: DeviceCapabilities, connectivity_graph: DiGraph
+) -> GateConnectivityCriterion:
+    raise NotImplementedError
+
+
+@create_gate_connectivity_criterion.register(IqmDeviceCapabilities)
+@create_gate_connectivity_criterion.register(RigettiDeviceCapabilities)
+def _(
+    properties: RigettiDeviceCapabilities, connectivity_graph: DiGraph
+) -> GateConnectivityCriterion:
+    """
+    Both IQM and Rigetti have undirected connectivity graphs; Rigetti device capabilities
+    provide back edges, but the calibration data only provides edges in one direction.
+    Additionally, IQM does not provide back edges in its connectivity_graph (nor is this
+    resolved manually by AwsDevice at the moment).
+    """
+    gate_connectivity_graph = connectivity_graph.copy()
+    edge_properties = properties.standardized.twoQubitProperties
+    for u, v in gate_connectivity_graph.edges:
+        edge_key = "-".join([str(qubit) for qubit in (u, v)])
+        edge_property = edge_properties.get(edge_key)
+        if not edge_property:
```

Comment:
Added a comment regarding this! This check is in case there is an edge between qubits in their connectivity graph, but the QHP has not provided calibration data about the gates that can be applied to the qubit-pair; the per-edge/per-two-qubit-gate calibration data is used to determine what gates can be applied to a qubit-pair.